src/test/java/com/google/cloud/tools/appengine/operations/DevServerJava9OrAboveTest.java

```diff
@@ -584,4 +893,31 @@ public void testGetGaeRuntimeJava_isJava8() {
   public void testGetGaeRuntimeJava_isNotJava8() {
     Assert.assertEquals("java7", DevServer.getGaeRuntimeJava(false));
   }
+
+  @Test
+  public void testGetJdkVersion_isValid() {
+    Assert.assertEquals(8, DevServer.getJdkMajorVersion("1.8"));
+    Assert.assertEquals(9, DevServer.getJdkMajorVersion("9"));
+    Assert.assertEquals(11, DevServer.getJdkMajorVersion("11"));
+    Assert.assertEquals(17, DevServer.getJdkMajorVersion("17"));
+    Assert.assertEquals(21, DevServer.getJdkMajorVersion("21"));
+
+    // This would be accepted for Java 8, but should expect them to be in
+    // `java.specification.version` syntax
+    Assert.assertEquals(8, DevServer.getJdkMajorVersion("1.8.0_181-google-v7"));
```

Comment:
What is the `java.specification.version` syntax and do we have any validation for it? What if an invalid value("1.8.abcd") being passed in?

Reply:
The syntax is just the result from the `java.specification.version` System Property. The only validation done is to check that it can be parsed into an integer. 

`1.8.abcd` -> 8
`1.11.abcd` -> 11

The two examples above are not "real" JDK versions and users can send in dummy values. I just use "best-effort" parsing to get the version and the expectation is that they would pass in a value with the format of 1.8 or 9, 10, 11, ... 21.